test/snapshot/widget_snapshot_checker.cc

```diff
@@ -0,0 +1,185 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/test/snapshot/widget_snapshot_checker.h"
+
+#include <string>
+
+#include "base/files/file_util.h"
+#include "base/memory/ref_counted_memory.h"
+#include "base/path_service.h"
+#include "base/threading/thread_restrictions.h"
+#include "brave/common/brave_paths.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "ui/compositor/test/draw_waiter_for_test.h"
+#include "ui/gfx/image/image.h"
+#include "ui/snapshot/snapshot.h"
+#include "ui/views/widget/widget.h"
+
+#if defined(USE_AURA)
+#include "ui/snapshot/snapshot_aura.h"
+#endif
+
+namespace {
+
+constexpr char kSnapshotFileName[] = "snapshot.png";
+
+base::StringPiece GetPlatformName() {
+#if defined(OS_WIN)
+  return "win";
+#elif defined(OS_MAC)
+  return "mac";
+#elif defined(OS_LINUX)
+  return "linux";
+#endif  // defined(OS_WIN)
+}
+
+bool IsSnapshotCheckingSupported() {
+  // TODO(https://github.com/brave/brave-browser/issues/17024): Add snapshots
+  // checking support for MacOS.
+#if defined(OS_MAC)
+  return false;
+#else
+  return true;
+#endif  // defined(OS_MAC)
+}
+
+bool WriteFailedSnapshotFile(scoped_refptr<base::RefCountedMemory> data,
+                             base::FilePath failed_snapshot_dir) {
+  base::ScopedAllowBlockingForTesting allow_block;
+  if (!base::CreateDirectoryAndGetError(failed_snapshot_dir, nullptr /*error*/)) {
+    return false;
+  }
+  const base::FilePath failed_snapshot_path =
+      failed_snapshot_dir.AppendASCII(kSnapshotFileName);
+
+  if (!base::WriteFile(failed_snapshot_path, data->front_as<char>(),
+                       data->size())) {
+    return false;
+  }
+
+  return true;
+}
+
+void Capture(views::Widget* widget, gfx::Image* image) {
+  // Wait for painting complete.
+  ui::Compositor* compositor = widget->GetCompositor();
+  ui::DrawWaiterForTest::WaitForCompositingEnded(compositor);
+
+#if defined(USE_AURA)
+  const gfx::Rect widget_bounds = widget->GetRootView()->bounds();
+  const auto on_got_snapshot = [](base::RunLoop* run_loop, gfx::Image* image,
+                                  gfx::Image got_image) {
+    *image = got_image;
+    run_loop->Quit();
+  };
+  base::RunLoop run_loop;
+  ui::GrabWindowSnapshotAsyncAura(
+      widget->GetNativeWindow(), widget_bounds,
+      base::BindOnce(on_got_snapshot, &run_loop, image));
+  run_loop.Run();
+#endif  // defined(USE_AURA)
+}
+
+bool CompareSnaphot(scoped_refptr<base::RefCountedMemory> png_bytes,
+                    base::FilePath snapshot_path) {
+  base::ScopedAllowBlockingForTesting allow_blocking;
+
+  if (!png_bytes->size()) {
+    return false;
+  }
+
+  std::string file_contents;
+  if (!base::ReadFileToString(snapshot_path, &file_contents)) {
+    return false;
+  }
+
+  if (file_contents.size() == 0) {
```

Comment:
`.empty()`

Reply:
Code was removed by moving to helper function from cc namespace: 
cc::MatchesPNGFile()